compiler/AST/LoopExpr.cpp

```diff
@@ -178,13 +178,13 @@ static CallExpr* buildLoopExprFunctions(LoopExpr* faExpr);
 static void addIterRecShape(CallExpr* forallExprCall,
                             bool parallel, bool zippered);
 
-class LowerLoopExprVisitor : public AstVisitorTraverse
+class LowerLoopExprVisitor final : public AstVisitorTraverse
 {
   public:
     LowerLoopExprVisitor() { }
-    virtual ~LowerLoopExprVisitor() { }
+    ~LowerLoopExprVisitor() { }
```

Comment:
You should use `override` for destructors too, to ensure that the base class destructor is `virtual`, e.g.
```C++
~LowerLoopExprVisitor() override { }
```
For trivial destructors, I prefer `= default`, which can be combined with `virtual`, `override` and `final`:
```C++
~LowerLoopExprVisitor() override = default;
```